From 57e97b7abad0b0b4c62f4b01c4c91674251243ab Mon Sep 17 00:00:00 2001
From: Yifeng Zhao <yifeng.zhao@rock-chips.com>
Date: Fri, 27 Mar 2020 10:53:25 +0800
Subject: [PATCH] drivers: mtd: nand: rockchip: fix the problem of wrong
 configuration for the timing

FMWAIT configured with a wrong value to cause timing error.

Signed-off-by: Yifeng Zhao <yifeng.zhao@rock-chips.com>
Change-Id: Ib09b79b5702bdff3010a578a50fc0d97854c03f1
---
 drivers/mtd/nand/raw/rockchip_nand_v6.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/nand/raw/rockchip_nand_v6.c b/drivers/mtd/nand/raw/rockchip_nand_v6.c
index 216557244039..5ff2063a214d 100644
--- a/drivers/mtd/nand/raw/rockchip_nand_v6.c
+++ b/drivers/mtd/nand/raw/rockchip_nand_v6.c
@@ -159,7 +159,7 @@ static void rk_nfc_init(struct rk_nfc *nfc)
 	writel(0, nfc->regs + NANDC_REG_V6_DMA_CFG);
 	writel(NANDC_V6_WP, nfc->regs + NANDC_REG_V6_FMCTL);
 	writel(NANDC_V6_FL_RST, nfc->regs + NANDC_REG_V6_FLCTL);
-	writel(0x1081, nfc->regs + NANDC_REG_V6_FMWAIT);
+	writel(0x1061, nfc->regs + NANDC_REG_V6_FMWAIT);
 }
 
 static irqreturn_t rk_nfc_interrupt(int irq, void *dev_id)
@@ -627,7 +627,7 @@ static int rk_nfc_setup_data_interface(struct mtd_info *mtd, int csline,
 	if (csline == NAND_DATA_IFACE_CHECK_ONLY)
 		return 0;
 
-	writel(0x10801, nfc->regs + NANDC_REG_V6_FMWAIT);
+	writel(0x1081, nfc->regs + NANDC_REG_V6_FMWAIT);
 
 	return 0;
 }
-- 
2.35.3

